admin/invoice.php

```diff
@@ -232,7 +232,7 @@ function couponpopupWindow(url) {
         echo '          <tr>' . "\n" .
              '            <td class="smallText" align="center" valign="top">' . zen_datetime_short($orders_history->fields['date_added']) . '</td>' . "\n";
         echo '            <td class="smallText" valign="top">' . $orders_status_array[$orders_history->fields['orders_status_id']] . '</td>' . "\n";
-        echo '            <td class="smallText" valign="top">' . ($orders_history->fields['comments'] == '' ? TEXT_NONE : nl2br(zen_db_output($orders_history->fields['comments']))) . '&nbsp;</td>' . "\n" .
+        echo '            <td class="smallText" valign="top">' . ($orders_history->fields['comments'] == '' ? TEXT_NONE : nl2br(zen_output_string_protected($orders_history->fields['comments']))) . '&nbsp;</td>' . "\n" .

```

Comment:
Just realized (perhaps a second time) that this substitution will provide inaccurate results in the current arrangement.  zen_db_output forces htmlspecialchars($var, ENT_COMPAT, CHARSET, TRUE) while the admin zen_output_string_protected forces htmlspecialchars($var, ENT_COMPAT, CHARSET, FALSE)... Thus, with this substitution (on the admin side at this time), the $var will not be "double" processed.. (I.e. &amp; will not become &amp;amp;)  This throws a spanner in the works, although could be instead hardcoded to be htmlspecialchars($var, ENT_COMPAT, CHARSET, TRUE);, then go through all of the admin code and basically swap the two sets of declarations because the FALSE version is used far less than the TRUE version seems to be... :) Then can figure out what it takes to convert the remaining FALSE versions instead... Just a thought...